cmd/installer/pkg/install.go

```diff
@@ -29,12 +30,21 @@ func Install(p runtime.Platform, config runtime.Configurator, sequence runtime.S
 	cmdline := kernel.NewCmdline("")
 	cmdline.Append("initrd", filepath.Join("/", "default", constants.InitramfsAsset))
 	cmdline.Append(constants.KernelParamPlatform, p.Name())
-	cmdline.Append(constants.KernelParamConfig, opts.ConfigSource)
 
 	if err = cmdline.AppendAll(p.KernelArgs().Strings()); err != nil {
 		return err
 	}
 
+	option := cmdline.Get(constants.KernelParamConfig).First()
+
+	switch {
+	case option == nil:
+		cmdline.Append(constants.KernelParamConfig, opts.ConfigSource)
+	case option != nil && opts.ConfigSource != "none":
+		fmt.Printf("warning: no kernel arguments supplied for %s\n", constants.KernelParamConfig)
+		fmt.Println("this may cause your machine to fail to boot")
+	}
+
```

Comment:
This should be in `cmd/installer/cmd/install.go`. The idea was that this package is a simple point and fire function, and that all the logic for figuring out the installation parameters would happen in `cmd/installer/cmd/install.go`.